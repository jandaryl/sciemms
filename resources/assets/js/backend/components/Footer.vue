<template>
  <Footer>
    <strong>Copyright &copy; {{ new Date().getFullYear() }} <a href="/" target="_blank">{{ name }}</a>.
    </strong>&nbsp;{{ $t('labels.all_rights_reserved') }}
  </Footer>
</template>
<script>
export default {
  name: 'AppFooter',
  props: {
    name: {
      type: String,
      default: ''
    },
    editorName: {
      type: String,
      default: ''
    },
    editorSiteUrl: {
      type: String,
      default: ''
    }
  }
}
</script>
