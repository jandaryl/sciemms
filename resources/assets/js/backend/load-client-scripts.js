/**
 * Tip:
 *     Sweetalert2        - for beautiful popup boxes.
 *     Flatpicker         - for elegant datetime picker.
 *     FlatpickrLocaleFr  - for datetime picker localization.
 */
import swal from 'sweetalert2'
import Flatpickr from 'flatpickr'
import FlatpickrLocaleFr from 'flatpickr/dist/l10n/fr'

/**
 * Store the JS plugins
 */
window.swal = swal
window.Flatpickr = Flatpickr
window.FlatpickrLocaleFr = FlatpickrLocaleFr

/**
 * JS Settings App
 *
 * Decoded the passed json config from backend.blade.php
 */
let jsonSettings = document.querySelector('[data-settings-selector="settings-json"]')
window.settings = jsonSettings ? JSON.parse(jsonSettings.textContent) : {}
