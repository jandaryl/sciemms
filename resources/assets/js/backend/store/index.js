/**
 * Use the Vue and Vuex
 */
import Vue from 'vue'
import Vuex from 'vuex'

/**
 * Call the actions and mutations functions.
 */
import { createActions } from './actions'
import mutations from './mutations'

/**
 * It will register the Vuex in the Vue instance.
 */
Vue.use(Vuex)

/**
 * Function that connect the whole logic of Vuex.
 * It will use actions and mutations functions.
 * Set the state and wait for AJAX data.
 * Update the counters properties.
 */
export function createStore (route) {
  /**
   * Call the action function.
   */
  const actions = createActions(route)

  return new Vuex.Store({
    state: {
      counters: {
        newPostsCount: 0,
        pendingPostsCount: 0,
        publishedPostsCount: 0,
        activeUsersCount: 0,
        formSubmissionsCount: 0
      }
    },
    actions,
    mutations
  })
}
