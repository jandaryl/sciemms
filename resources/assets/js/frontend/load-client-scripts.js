/**
 * Tip :
 *      Fontawesome           - for icons and brands.
 *      Slick Carousel        - for easy and nice carousel.
 *      Intl-tel-input        - for phone number validate and management.
 *      Pwstrength-bootstrap  - for the awesome password meter and strength.
 *      Sweetalert2           - for the beautiful popup boxes it used to notify.
 *      WebFont               - for @font-face manager that handle different sources.
 *      Turbolinks            - for making the web navigating become faster.
 */
import { fontawesome } from '../fontawesome'
import 'slick-carousel'
import 'intl-tel-input'
import 'pwstrength-bootstrap/dist/pwstrength-bootstrap'
import swal from 'sweetalert2'
import WebFont from 'webfontloader'
import Turbolinks from 'turbolinks'

/**
 * Store the json settings in window to have global access.
 */
let jsonSettings = document.querySelector('[data-settings-selector="settings-json"]')
window.settings = jsonSettings ? JSON.parse(jsonSettings.textContent) : {}

/**
 * Store the swal and language from html in window.
 */
window.swal = swal
window.locale = $('html').attr('lang')

/**
 * Function that will used to initialize the plugins.
 * It will used in the frontend User Interface.
 */
export default (createApp) => {
  /**
   * Start the Turbolinks
   */
  Turbolinks.start()

  /**
   * Load the web font that will be used.
   */
  WebFont.load({
    google: {
      families: ['Roboto']
    }
  })

  /**
   * Add event listener when load the app.
   * It will listen to cookie consent from initialization.
   */
  window.addEventListener('load', () => {
    window.cookieconsent.initialise({
      'palette': {
        'popup': {
          'background': '#fff',
          'text': '#777'
        },
        'button': {
          'background': '#3097d1',
          'text': '#ffffff'
        }
      },
      'showLink': false,
      'theme': 'edgeless',
      'content': {
        'message': window.settings.cookieconsent.message,
        'dismiss': window.settings.cookieconsent.dismiss
      }
    })
  })

  /**
   * Add event listener for turbo links load.
   */
  document.addEventListener('turbolinks:load', () => {
    /**
     * Check if the app is defined in the HTML.
     * Then mount the Vue.
     */
    if (document.getElementById('app') !== null) {
      const {app} = createApp()
      app.$mount('#app')
    }
    /**
     * Bind all bootstrap tooltips.
     */
    $('[data-toggle="tooltip"]').tooltip()
    /**
     * Bind all bootstrap popovers.
     */
    $('[data-toggle="popover"]').popover()
    /**
     * Set the font awesome.
     */
    fontawesome.dom.i2svg()

    /**
     * Set the configs of Slick plugin.
     */
    $('[data-toggle="slider"]')
      .not('.slick-initialized')
      .slick({
        dots: true,
        infinite: true,
        speed: 300,
        slidesToShow: 3,
        slidesToScroll: 3,
        responsive: [
          {
            breakpoint: 768,
            settings: {
              slidesToShow: 2
            }
          },
          {
            breakpoint: 576,
            settings: {
              slidesToShow: 1
            }
          }
        ]
      })

    /**
     * Bind all the Swal to confirm buttons.
     */
    $('[data-toggle="confirm"]').click((e) => {
      /**
       * Prevent the default events.
       */
      e.preventDefault()

      /**
       * Define the Swal settings.
       */
      window.swal({
        title: $(e.currentTarget).attr('data-trans-title'),
        type: 'warning',
        showCancelButton: true,
        cancelButtonText: $(e.currentTarget).attr('data-trans-button-cancel'),
        confirmButtonColor: '#dd4b39',
        confirmButtonText: $(e.currentTarget).attr('data-trans-button-confirm')
      }).then((result) => {
        if (result.value) {
          $(e.target).closest('form').submit()
        }
      })
    })

    /**
     * Bind the password strength meter then define the config.
     */
    $('[data-toggle="password-strength-meter"]').pwstrength({
      ui: {
        bootstrap4: true
      }
    })

    /**
     * Bind the intl Tel Input to type "tel" then define the config.
     */
    $('[type="tel"]').intlTelInput({
      autoPlaceholder: 'aggressive',
      utilsScript: 'https://cdnjs.cloudflare.com/ajax/libs/intl-tel-input/11.0.14/js/utils.js',
      initialCountry: window.locale === 'en' ? 'us' : window.locale,
      preferredCountries: ['us', 'gb', 'fr']
    })

    /**
     * Bootstrap tabs nav specific hash manager.
     */
    let hash = document.location.hash
    let tabanchor = $('.nav-tabs a:first')

    if (hash) {
      tabanchor = $(`.nav-tabs a[href="${hash}"]`)
    }

    tabanchor.tab('show')

    $('a[data-toggle="tab"]').on('show.bs.tab', (e) => {
      window.location.hash = e.target.hash
    })
  })
}
