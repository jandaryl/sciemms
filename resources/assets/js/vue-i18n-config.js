/**
 * Vue Instance
 */
import Vue from 'vue'

/**
 * Vue Translator & generated locales
 */
import VueI18n from 'vue-i18n'
import Locales from './vue-i18n-locales.generated.js'

/**
 * Use the VueI18n as internationalization plugin.
 */
Vue.use(VueI18n)

/**
 * The createLocales function will set the passed locale value from config app,
 * and set also the vue generated locales that will be correspond to translated data.
 */
export function createLocales (locale) {
  return new VueI18n({
    locale: locale,
    messages: Locales
  })
}
