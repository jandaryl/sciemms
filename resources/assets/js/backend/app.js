/**
 * Backend Client Window Settings
 */
import './load-client-scripts'

/**
 * Vue & Axios with config.
 */
import Vue from 'vue'
import '../axios-config'

/**
 * Babel-polyfill & Bootstrap Vue
 */
import 'babel-polyfill'
import BootstrapVue from 'bootstrap-vue/dist/bootstrap-vue.esm'

/**
 * Vendor plugins components.
 */
import '../../vendor/coreui/components'
import DataTable from './components/Plugins/DataTable'
import RichTextEditor from './components/Plugins/RichTextEditor'
import DateTimePicker from './components/Plugins/DateTimePicker'
import Switch from './components/Plugins/Switch'
import vSelect from './components/Plugins/Select'

/**
 * Vue helper functions.
 */
import { createRouter } from './router'
import { createStore } from './store'
import { createLocales } from '../vue-i18n-config'

/**
 * Main Component
 */
import App from './App.vue'

/**
 * Notification Tool
 */
import Noty from 'noty'

/**
 * Use Bootstrap Vue
 */
Vue.use(BootstrapVue)

/**
 * Use custom components.
 */
Vue.component('v-select', vSelect)
Vue.component('c-switch', Switch)
Vue.component('p-datetimepicker', DateTimePicker)
Vue.component('p-richtexteditor', RichTextEditor)
Vue.component('b-datatable', DataTable)

/**
 * The create app function will use to render into admin dashboard.
 */
export function createApp () {
  /**
   * The create locales will get the locale value from windows that define json settings.
   *
   * Ex. createLocales('en')
   */
  const i18n = createLocales(window.settings.locale)

  /**
   * The create router will get the admin prefix path that define from config app.
   *
   * Ex. createRouter('admin', 'en')
   */
  const router = createRouter(window.settings.adminHomePath, i18n)

  /**
   * The create store will get the named routes that provide by @routes()
   *
   * Ex. createStore('admin.posts.draft.counter')
   */
  const store = createStore(window.route)

  /**
   * Server-side settings
   *
   * To access it as global in all the json settings from load-client-scripts.js
   *
   * Ex. $app.locale, $app.user
   */
  Vue.prototype.$app = window.settings

  /**
   * Server-side named routes
   *
   * Get the named routes from the ziggy function.
   *
   * Ex. @routes()
   */
  Vue.prototype.$app.route = window.route

  /**
   * Client-side permissions
   *
   * 1. Check the user instance if true that came from window settings.
   * 2. Check the user if the id is equal 1 correspond to super admin.
   * 3. Return true or the permission if it is includes from window settings.
   */
  if (Vue.prototype.$app.user) {
    Vue.prototype.$app.user.can = (permission) => {
      if (Vue.prototype.$app.user.id === 1) {
        return true
      }
      return Vue.prototype.$app.permissions.includes(permission)
    }
  }

  /**
   * Form Object to Form Data Object converter
   *
   * The idea is the object is came from Form's Vue.
   * Where it used to act as progressively in vue instance.
   * Object is specifically in model property of data method of vue.
   * Where it was stored the form data as object with form properties.
   *
   * Tip :
   *      Form Data Object use to send form data by using ajax request.
   *
   * 1. Create an new Form Data Object.
   * 2. Separate the property from object.
   * 3. Check if it is a null, append the no value.
   * 4. Check if it is a boolean, append the 1 or 0 value.
   * 5. Check if it is a Date append the converted date value.
   * 6. Check if it is a Object append the property as form key.
   * 7. Store it in the global variable for send request to the backend.
   */
  let objectToFormData = (object, form, namespace) => {
    let formData = form || new FormData()

    for (let property in object) {
      if (!object.hasOwnProperty(property)) {
        continue
      }

      let formKey = namespace ? `${namespace}[${property}]` : property

      if (object[property] === null) {
        formData.append(formKey, '')
        continue
      }
      if (typeof object[property] === 'boolean') {
        formData.append(formKey, object[property] ? '1' : '0')
        continue
      }
      if (object[property] instanceof Date) {
        formData.append(formKey, object[property].toISOString())
        continue
      }
      if (typeof object[property] === 'object' && !(object[property] instanceof File)) {
        objectToFormData(object[property], formData, formKey)
        continue
      }
      formData.append(formKey, object[property])
    }

    return formData
  }

  /**
   * Store the objectToFormData into global variable.
   * To used in the form mixins for sending form request.
   */
  Vue.prototype.$app.objectToFormData = objectToFormData

  /**
   * Set the configs and define function for notification in the dashboard.
   */
  let noty = (type, text) => {
    new Noty({
      layout: 'topRight',
      theme: 'bootstrap-v4',
      timeout: 2000,
      text,
      type
    }).show()
  }

  /**
   * Get the state from the form mixins to notify the user.
   * It might be alert, success, error, warning, and info.
   */
  Vue.prototype.$app.noty = {
    alert: (text) => {
      noty('alert', text)
    },
    success: (text) => {
      noty('success', text)
    },
    error: (text) => {
      noty('error', text)
    },
    warning: (text) => {
      noty('warning', text)
    },
    info: (text) => {
      noty('info', text)
    }
  }

  /**
   * Notify the user if there is an error in the system.
   * It could be 403, undefined, or general exception.
   */
  Vue.prototype.$app.error = (error) => {
    if (error instanceof String) {
      noty('error', error)
      return
    }

    if (error.response) {
      if (error.response.status === 403) {
        noty('error', i18n.t('exceptions.unauthorized'))
        return
      }

      if (error.response.data.error !== undefined) {
        noty('error', error.response.data.message)
        return
      }
    }

    noty('error', i18n.t('exceptions.general'))
  }

  /**
   * Global router guards use to override the HTML title with meta label or app name.
   *
   * Ex. <title>Hello World!</title>
   */
  router.beforeEach((to, from, next) => {
    document.title = `${to.meta.label} | ${window.settings.appName}`
    next()
  })

  /**
   * Make new vue instance with router, store, i18n.
   * Then render it into App component.
   */
  const app = new Vue({
    router,
    store,
    i18n,
    render: (h) => h(App)
  })

  return {app, router, store}
}

/**
 * Check if there is an ID of app that defined in the html.
 * Then mount the vue components.
 */
if (document.getElementById('app') !== null) {
  const {app} = createApp()
  app.$mount('#app')
}
