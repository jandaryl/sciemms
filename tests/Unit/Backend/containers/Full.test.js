import Vue from "vue"
import '../Utils/helpers/unit-test-wrappers'
import Full from "@/backend/containers/Full.vue"
import BootstrapVue from 'bootstrap-vue/dist/bootstrap-vue.esm'
import Breadcrumb from '@vendor/components/Breadcrumb/Breadcrumb'
import { shallowMount } from "@vue/test-utils"
import { createRouter } from "@/backend/router"
import { createLocales } from "@/vue-i18n-config"
import { windowRoute } from "../Utils/stubs/load-client-scripts"
import { SidebarFooter, SidebarMinimizer, SidebarNav, Sidebar } from '@vendor/components/Sidebar'

let bootstrapVue = Vue.use(BootstrapVue)
let i18n = createLocales('en')
let router = createRouter('admin', i18n)

Vue.component('Breadcrumb', Breadcrumb)
Vue.component('SidebarMinimizer', SidebarMinimizer)
Vue.component('SidebarFooter', SidebarFooter)
Vue.component('SidebarNav', SidebarNav)
Vue.component('Sidebar', Sidebar)

let what = 'is this code'
let name = 'Jan Daryl Galbo'
let route = "test\.route"
let routeName = 'Test Route'
let routeMatched = true
let supername  = 'superadmin'
let appName  = 'SCIEMMS'
let editorName = 'Vue Test'
let editorSiteUrl = 'https://sciemms.frb.io'
let testLocale = 'Test Locale'
let dummyData = { sync: false,
      i18n,
      router,
      bootstrapVue ,
      mocks: {
        $app: {
          user: {
            name: name,
            can: () => true
          },
          route: (route) => route,
          usurperName: supername,
          appName: appName,
          editorName: editorName,
          editorSiteUrl: editorSiteUrl
        },
        $store : {
          state: {
            counters: {
                newPostsCount: 0,
                pendingPostsCount: 0,
                publishedPostsCount: 0,
                activeUsersCount: 0,
                formSubmissionsCount: 0
              }
          }
        }
      }}
let App = (values = {}) => { return shallowMount(Full, values) }
let wrapper = App(dummyData)

let assertTrue = data => { expect(data).toBeTruthy() }
let assertSnapMatched = element => { expect(element).toMatchSnapshot() }

describe("Full Component", () => {
  it("is a vue instance", () => {

    assertTrue(wrapper.isVueInstance())
  })

  it("is matched in the previos snapshots", () => {

    assertSnapMatched(wrapper.element)
  })
})
