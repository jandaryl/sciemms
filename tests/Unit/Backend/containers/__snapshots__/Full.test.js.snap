// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full Component is matched in the previos snapshots 1`] = `
<div
  class="app"
>
  <appheader-stub />
   
  <div
    class="app-body"
  >
    <sidebar-stub
      fixed="true"
    >
      <div
        class="sidebar-header"
      >
        <i
          class="fe fe-user"
        />
          Jan Daryl Galbo
      
      </div>
       
      <sidebarnav-stub
        navitems="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      />
       
      <sidebarfooter-stub />
       
      <sidebarminimizer-stub />
    </sidebar-stub>
     
    <main
      class="main"
    >
      <balert-stub
        class="alert-top mb-0"
        dismisslabel="Close"
        variant="warning"
      >
        <span>
          You are actually logged as 
          <strong>
            Jan Daryl Galbo
          </strong>
          , you can logout as 
          <a
            data-turbolinks="false"
            href="admin.logout"
          >
            superadmin
          </a>
          .
        </span>
      </balert-stub>
       
      <balert-stub
        class="alert-top mb-0"
        dismisslabel="Close"
        show="true"
        variant="warning"
      >
        <span>
          Your account will be in limited mode as long as your email remains not confirmed. 
          <a
            href="user.confirmation.send"
          >
            Click here
          </a>
           in order to resend mail confirmation.
        </span>
      </balert-stub>
       
      <breadcrumb-stub
        list="[object Object],[object Object]"
      />
       
      <div
        class="container-fluid"
      >
        <router-view-stub
          name="default"
        />
      </div>
    </main>
     
    <aside
      fixed=""
    />
  </div>
   
  <appfooter-stub
    editorname="Vue Test"
    editorsiteurl="https://sciemms.frb.io"
    name="SCIEMMS"
  />
</div>
`;
